### ISSUE TITLE
Netdisco compatability with H3C wireless controller

### ISSUE THREAD

**Issue by ciscolive** [netdisco/netdisco#1047]
Currently, Netdisco does not support automatic discovery of Access Points (APs) and signal information related to H3C wireless controllers.  Dear team coulud you give me some advices ? thanks

**Comment by rc9000** [comment]
Yes unfortunately it looks like there is no support for H3C wireless at all. I'm haven't done much with the wireless parts of Netdisco and know zero about H3C, but for a rough outline:

 * The key bits are in lib/App/Netdisco/Worker/Plugin/Discover/Wireless.pm. There you can see various SNMP::Info calls like `$snmp->i_ssidmac` and `$snmp->i_80211channel` being made. 
 * If h3c has MIBS for wireless information, just implementing + overriding of the SNMP::Info methods can get you there.
 * If not and you need to use an API or CLI commands, you can implement a custom discovery worker that can do almost anything, see https://github.com/netdisco/netdisco/wiki/Worker-Cookbook for that.
 * In the end all that matters if that the data ends up in these database tables:
  * `device_port_ssid`
  * `device_port_wireless`
  * `node_wireless` (client information, this is filled by macsuck usually)

The difficulty with Netdisco development is that not only it needs motivated people to write the code, they also need to care about and have access to the hardware in question. Obviously that coincidence hasn't happened for H3C yet, but if you want to give it a try I'm sure you'd get plenty of support here.
